test/test_md5.rb

```diff
@@ -38,16 +44,30 @@ def test_hexdigest_nil
 
   def test_hexdigest_array
     assert_equal "4410ec34d9e6c1a68100ca0ce033fb17", MD5.hexdigest([])
-    assert_equal "759f28c1d0c20c22e79c91d339855d95", MD5.hexdigest([1])
-    assert_equal "9efe07b352dd94cd1cdc9d1a8d054f8f", MD5.hexdigest([1, 2])
-    assert_equal "60b1ef7201404f20c3d12f47499c3a1f", MD5.hexdigest([1, 2, 3])
-    assert_equal "2f82cdc166616208077ff1dd0a8faeff", MD5.hexdigest([1, 2, [3]])
+    # Ruby 2.4.0 merged Bignum and Fixnum into Integer which means we get different digests
+    if RUBY_VERSION >= "2.4.0"
```

Comment:
This string compare will fail in some common cases:

```
>> "2.3.2" > "2.3.10"
=> true
```
You can use `Gem::Version` to work around that:

```
>> Gem::Version.new("2.3.2") > Gem::Version.new("2.3.10")
=> false
```